engineering_tools/carma

```diff
@@ -204,7 +204,26 @@ carma-config__install() {
     echo "Cleaning up temporary container..."
     docker rm carma-config-tmp
 }
+carma__record() {
+    last_modified_file="$(find "/opt/carma/logs" -type f \( -name \*.bag.active -o -name \*.bag \) -printf "%T@ %p\0" | awk 'BEGIN {RS="\0";} {if (NR==1){minmtime=$1; $1=""; lastmodified=$0;} else if (minmtime<$1){minmtime=$1; $1=""; lastmodified=$0;}} END{print substr(lastmodified,2)}')"
 
+    printf "The last modified file is named %s\n" "${last_modified_file}"
+
+    if [[ ${last_modified_file} == *.bag.active ]]; then
+        rosbag reindex ${last_modified_file}
+
+        rosbag_name="$(basename -s .bag.active ${last_modified_file})"
+
+        if test -f "/opt/carma/logs/${rosbag_name}.bag.orig.active"; then
+            rm "/opt/carma/logs/${rosbag_name}.bag.orig.active"
+        fi
+
+        mv ${last_modified_file} "/opt/carma/logs/${rosbag_name}.bag"
+
+        printf "Created a rosbag named %s\n" "/opt/carma/logs/${rosbag_name}.bag"
+
+    fi
```

Comment:
What happens if no bag is found. There needs to be some sort of notice to the user